HangoverAvalonia and other fixes

Mbucari opened the pull request:
- Made HangoverAvalonia project, so now chardonnay releases are 100% avalonia and contain all Libation programs.
- Moved IsWindows, IsLinux and IsMacOS into LibationScaffolding.
- Disable mp3 options for mac builds.  I'll need to make yet another aac decoder version for mac!
- Changed mac and linux builds to standalone.  Now users don't have to install the dotnet runtime. I tested the mac build on Big Sur (only version I have available) and on Ubuntu 18 and 22.  It works!
- Fixed bug with Libation installation.  Avalonia's desktop lifetime is based around one main window. If the main window is closed, the application exits.  So running the setup routine requires chaining main windows, changing the main window to the next window in the chain BEFORE the current window is closed. (eg. `SetupDialog` > `LibationFilesDialog` > `SettingsDialog` > `MainWindow`)

**New publish commands for chardonnay**

**Windows**
```console
dotnet clean   -c Release
dotnet publish -c Release -o bin\Release\win-chardonnay   LibationAvalonia\LibationAvalonia.csproj -p:PublishProfile=LibationAvalonia\Properties\PublishProfiles\WindowsProfile.pubxml
dotnet publish -c Release -o bin\Release\win-chardonnay   LibationCli\LibationCli.csproj           -p:PublishProfile=LibationCli\Properties\PublishProfiles\WindowsProfile.pubxml
dotnet publish -c Release -o bin\Release\win-chardonnay   HangoverAvalonia\HangoverAvalonia.csproj -p:PublishProfile=HangoverAvalonia\Properties\PublishProfiles\WindowsProfile.pubxml
```
**Linux**
```console
dotnet clean   -c Release
dotnet publish -c Release -o bin\Release\linux-chardonnay LibationAvalonia\LibationAvalonia.csproj -p:PublishProfile=LibationAvalonia\Properties\PublishProfiles\LinuxProfile.pubxml
dotnet publish -c Release -o bin\Release\linux-chardonnay LibationCli\LibationCli.csproj           -p:PublishProfile=LibationCli\Properties\PublishProfiles\LinuxProfile.pubxml
dotnet publish -c Release -o bin\Release\linux-chardonnay HangoverAvalonia\HangoverAvalonia.csproj -p:PublishProfile=HangoverAvalonia\Properties\PublishProfiles\LinuxProfile.pubxml
```
**Mac**
```console
dotnet clean   -c Release
dotnet publish -c Release -o bin\Release\macos-chardonnay LibationAvalonia\LibationAvalonia.csproj -p:PublishProfile=LibationAvalonia\Properties\PublishProfiles\MacOSProfile.pubxml
dotnet publish -c Release -o bin\Release\macos-chardonnay LibationCli\LibationCli.csproj           -p:PublishProfile=LibationCli\Properties\PublishProfiles\MacOSProfile.pubxml
dotnet publish -c Release -o bin\Release\macos-chardonnay HangoverAvalonia\HangoverAvalonia.csproj -p:PublishProfile=HangoverAvalonia\Properties\PublishProfiles\MacOSProfile.pubxml
```

rmcrackan commented on Source/AppScaffolding/LibationScaffolding.cs:
```diff
@@ -26,6 +25,16 @@ public enum ReleaseIdentifier
 
 	public static class LibationScaffolding
 	{
+		public static readonly bool IsWindows;
```
I admit that it made sense to do it at compile time. The code is much more readable this way though

rmcrackan commented:
> Changed mac and linux builds to standalone. Now users don't have to install the dotnet runtime. I tested the mac build on Big Sur (only version I have available) and on Ubuntu 18 and 22. It works!

That's huge!

rmcrackan commented:
> New publish commands for chardonnay

Perfect, thanks. This is why I wanted to fix my deploy tool the right way when this all got shaken up recently. Adding the clean steps was a 1-liner. And including Hangover is just adding 'Hangover' to the projects list for mac and linux

rmcrackan commented:
Last comment/question: it sounds like the chardonnay install bug fix is major enough to justify a new version, correct?

rmcrackan commented:
After the pull, I see HangoverAvalonia in my files but it's not in the solution so I added it. In VS, HangoverAvalonia has a folder 'Models' which shows up with a red X. It doesn't keep it from building correctly but something does look amiss.

(I also updated the README regarding Hangover not being available)

Mbucari commented:
> it sounds like the chardonnay install bug fix is major enough to justify a new version, correct?

I think so. I was going to suggest you only release the Chardonnay builds because classic is unchanged, but you really can't do that because the old classic builds will see the Chardonnay update and not know the difference.

Mbucari commented:
> After the pull, I see HangoverAvalonia in my files but it's not in the solution so I added it.

Oops. I've gotten in the habit of not committing changes to the solution because I have a bunch of other stuff added in my debug environment.

 > In VS, HangoverAvalonia has a folder 'Models' which shows up with a red X. It doesn't keep it from building correctly but something does look amiss.

That's a default folder added by the Avalonia project template. You can just delete it.

> (I also updated the README regarding Hangover not being available)

Thanks!

rmcrackan commented:
> > (I also updated the README regarding Hangover not being available)

> Thanks!

No problem. As you can see, I'm not afraid of hard work :)

My neuroses of course has me self-conscious that you're doing so much right now and I haven't been. I have to remind myself that this is always the deal -- aside from non-beta bugs, everyone should do as much or as little as they feel like. Don't ever feel obligated though. My philosophy is that I do it when it's fun and I pause when it feels like work. That disclaimer out of the way, I'm very grateful for your creations lately. They've been tremendous.

Mbucari commented:
>  I have to remind myself that this is always the deal, everyone should do as much or as little as they feel like.

Absolutely! I have no resentments because I'm doing it by choice. It's a hobby (and possibly a resume builder).

> My philosophy is that I do it when it's fun and I pause when it feels like work.

And in that spirit, there's a list of things that need to be done that I just really don't feel like doing.  I'm going to make an issue with the list of TODOs, and maybe you can add the "help wanted" tag.

> I'm very grateful for your creations lately. They've been tremendous.

Thank you! I'm really pleased with how everything is coming together. Avalonia was hard at first, but now I think it's even easier to whip up a new form in it than in Winforms.  I'm going to write a little primer for working with AvaloniaUI and how bindings work so you can get in the groove.

rmcrackan commented:
> I'm going to write a little primer for working with AvaloniaUI and how bindings work so you can get in the groove.

Gracias
